File: springboot-remoting-server/src/main/java/com/example/remoting/service/impl/DnsServiceJmsImpl.java
package com.example.remoting.service.impl;

import com.example.remoting.service.DnsService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.beans.factory.annotation.Qualifier;
import org.springframework.stereotype.Service;

@Service
@Qualifier(value = "DnsServiceJmsImpl")
public class DnsServiceJmsImpl implements DnsService {
    @Autowired
    BusinessLogic businessLogic;

    @Override
    public String getName(String ip) {

        return businessLogic.getName(ip);
    }
}

File: springboot-remoting-kotlin-server/src/main/kotlin/com/example/remoting/service/impl/DnsServiceJmsImpl.kt
package com.example.remoting.service.impl

import com.example.remoting.service.DnsService
import org.slf4j.LoggerFactory
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.beans.factory.annotation.Qualifier
import org.springframework.stereotype.Service

@Service
@Qualifier(value = "DnsServiceJmsImpl")
class DnsServiceJmsImpl : DnsService {
    private val log = LoggerFactory.getLogger(DnsServiceJmsImpl::class.java)


    @Autowired
    lateinit var businessLogic: BusinessLogic

    override fun getName(ip: String): String {
        var name: String = businessLogic.getName(ip)
        log.info("Jms remoting method called, return value: {}", name)

        return name
    }
}